src/nested_pandas/nestedframe/core.py

```diff
@@ -258,7 +258,17 @@ def _getitem_list(self, item):
         return result
 
     def __setitem__(self, key, value):
-        """Adds custom __setitem__ behavior for nested columns"""
+        """Custom __setitem__ for NestedFrame: auto-nest DataFrame assignment to new columns."""
```

Comment:
Since there's not a nice docstring to have an example of this in, it would be nice to have a quick example in the docs, probably around here: https://nested-pandas.readthedocs.io/en/latest/tutorials/data_manipulation.html#Adding-New-Nested-Structures